src/OpenSage.Game/Logic/AI/AIStates/ChinookCombatDropState.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.Collections.Generic;
+using OpenSage.Logic.Object;
+using OpenSage.Mathematics;
+
+namespace OpenSage.Logic.AI.AIStates;
+
+internal sealed class ChinookCombatDropState : State
+{
+    private readonly GameObject _gameObject;
+    private readonly List<Rope> _ropes = new();
+
+    public ChinookCombatDropState(GameObject gameObject)
+    {
+        _gameObject = gameObject;
+    }
+
+    public override void OnEnter()
+    {
+        ResetRopes(((ChinookAIUpdate)_gameObject.AIUpdate).ModuleData.NumRopes);
+    }
+
+    public override void OnExit()
+    {
+        _ropes.Clear();
+    }
+
+    public override void Persist(StatePersister reader)
+    {
+        reader.PersistVersion(2);
+
+        var numRopes = _ropes.Count;
+        reader.PersistInt32(ref numRopes);
+
+        if (reader.Mode == StatePersistMode.Read)
+        {
+            ResetRopes(numRopes);
+        }
+
+        reader.BeginArray("Ropes");
+        foreach (var rope in _ropes)
+        {
+            reader.PersistObject(rope);
+        }
+        reader.EndArray();
```

Comment:
test this, but I think it should work
```suggestion
        reader.PersistListWithUInt32Count(_ropes, (StatePersister persister, ref Rope item) =>
        {
            item ??= new Rope();
            persister.PersistObjectValue(item);
        });
```

Reply:
Thanks! https://github.com/OpenSAGE/OpenSAGE/pull/953/commits/daabe241f40c986df4012dabdbc9f0289655aebc